src/main/kotlin/com/jetbrains/snakecharm/codeInsight/completion/SmkRulesAndCheckpointsCompletionContributor.kt

```diff
@@ -87,9 +87,17 @@ private class SmkRulesAndCheckpointsObjectsCompletionProvider : CompletionProvid
     ) {
         val variants = collectVariantsForElement(parameters.position)
 
+        val parentElement = parameters.position.parent
+        val referencedObject = parameters.withPosition(parentElement, parentElement.textOffset).originalPosition?.parent
+        if (referencedObject is PyReferenceExpression) {
+            when (referencedObject.name) {
+                SnakemakeNames.SMK_VARS_RULES -> variants.removeAll { it.second is SmkCheckPoint }
```

Comment:
Not clear, why we collect both rules and checkpoints from an index in line 88 and then remove rules or checkpoints here. Why not to collect the only required type of elements from the index? Index keys differ for rules and checkpoints.

Reply:
done